main/src/com/google/refine/commands/workspace/GetMemoryInfoCommand.java

```diff
@@ -0,0 +1,37 @@
+
+package com.google.refine.commands.workspace;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.refine.commands.Command;
+import oshi.SystemInfo;
+import oshi.hardware.GlobalMemory;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class GetMemoryInfoCommand extends Command {
+
+    private static final GlobalMemory systemMemory = new SystemInfo().getHardware().getMemory();
+
+    public static class MemoryInfo {
+
+        @JsonProperty("available_memory")
+        protected Long availableMemory = systemMemory.getAvailable();
+
+        @JsonProperty("total_memory")
+        protected Long totalMemory = systemMemory.getTotal();
```

Comment:
I am a bit suspicious about those calls. Given their description it seems that they refer to the total memory available and used by the system. This is not what we should be watching out for: rather, we should check how much memory our JVM is using, and how much was allocated to it.

I am hopeful that this information can be accessed in pure Java, without the need for a dependency like `oshi`.

Reply:
Yes, it can